pkg/tsdbctl/check.go

```diff
@@ -91,58 +126,172 @@ func (cc *checkCommandeer) check() error {
+		}
+	}
+	if cc.to == "" {
+		to = math.MaxInt64
+	} else {
+		to, err = utils.Str2unixTime(cc.to)
+		return nil, errors.Wrap(err, "failed to parse end time")
+	}
+	partitionInterval, _ := utils.Str2duration(schema.PartitionSchemaInfo.PartitionerInterval)
+	var respChans []chan *v3io.Response
+	for _, partition := range schema.Partitions {
+		partitionEndTime := partition.StartTime + partitionInterval
+		if partitionEndTime >= from && partition.StartTime <= to {
+			respChan := make(chan *v3io.Response, 1)
+			respChans = append(respChans, respChan)
+			objName := fmt.Sprintf("%s_%d.%x", cc.metricName, hash%uint64(numBuckets), hash)
+			objPath := path.Join(tablePath, strconv.FormatInt(partition.StartTime/1000, 10), objName)
+			input := v3io.GetItemInput{Path: objPath, AttributeNames: []string{"**"}}
+			_, err := container.GetItem(&input, nil, respChan)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to send GetItem request")
+			}
+		}
 	}
+	return respChans, nil
+}
+
+func (cc *checkCommandeer) printResponse(resp *v3io.Response) error {
 
 	// Print the metric metadata
 	item := resp.Output.(*v3io.GetItemOutput).Item
-	objName, _ := item.GetFieldString("__name")
+	objPath := resp.Request().Input.(*v3io.GetItemInput).Path
 	metricName, _ := item.GetFieldString("_name")
 	lsetString, _ := item.GetFieldString("_lset")
 	maxtime, _ := item.GetFieldInt("_maxtime")
-	fmt.Printf("Metric Item: %s,  %s {%s}  maxtime: %d\n", objName, metricName, lsetString, maxtime)
+	fmt.Printf("Metric Item: %s,  %s{%s}  maxtime: %d\n", objPath, metricName, lsetString, maxtime)
 
 	// Decompress and print metrics
-	for _, attr := range cc.attrs {
+	for attr, values := range item {
+		if strings.HasPrefix(attr, "_v") {
+			fmt.Println("\tAttr:", attr)
 
-		values := item.GetField(attr)
-		fmt.Println("Attr:", attr)
-
-		if values != nil {
-			bytes := values.([]byte)
-			chunk, err := chunkenc.FromData(cc.rootCommandeer.logger, chunkenc.EncXOR, bytes, 0)
-			if err != nil {
-				cc.rootCommandeer.logger.ErrorWith("Error reading chunk buffer.", "Lset", lset, "err", err)
-				return err
-			} else {
-				count := 0
-				iter := chunk.Iterator()
-				for iter.Next() {
-					t, v := iter.At()
-					tstr := time.Unix(int64(t/1000), 0).UTC().Format(time.RFC3339)
-					fmt.Printf("Unix timestamp=%d, t=%s, v=%.4f \n", t, tstr, v)
-					count++
-				}
-				if iter.Err() != nil {
-					return errors.Wrap(iter.Err(), "Failed to read the iterator.")
-				}
-
-				compressionRatio := 0.0
-				bytesCount := len(bytes)
-				if count > 0 {
-					compressionRatio = float64(bytesCount) / float64(count)
+			if values != nil {
+				bytes := values.([]byte)
+				err := cc.printValues(bytes)
+				if err != nil {
+					return err
 				}
-				fmt.Printf("Total size=%d, Count=%d, Compression ratio=%.2f\n",
-					bytesCount, count, compressionRatio)
 			}
 		}
 	}
+	return nil
+}
 
+func (cc *checkCommandeer) printValues(bytes []byte) error {
```

Comment:
Consider moving useful helper functions to a package so they might be reused by others

Reply:
That's not a generic utility, just some specific logic.